src/components/app/custom-els/FileDrop/index.ts

```diff
@@ -0,0 +1,144 @@
+import * as styles from './styles.css';
+
+class DragState {
+
+  private element: FileDrop;
+  private hasDragData: boolean;
```

Comment:
I don't think we need this. Just call `getMatchingFiles` each time you need to know. If this becomes a performance issue (which I seriously doubt) we can look at a way to fix it.

Reply:
It really felt wasteful, and also I don't have the actual Data until we're about to drop the file.